azurelinuxagent/pa/rdma/suse.py

```diff
@@ -93,8 +93,7 @@ def install_driver(self):
                 msg = 'RDMA: Successfully installed "%s" from '
                 msg += 'configured repositories'
                 logger.info(msg % complete_name)
-                self.load_driver_module()
-                if requires_reboot:
+                if not self.load_driver_module() or requires_reboot:
```

Comment:
It will not trigger the next reboot, because the driver package is already before the next reboot. On the next reboot, we'll simply return not doing anything:

                if fw_version in version and installed == 'i':
                    info_msg = 'RDMA: Matching driver package "%s-%s" '
                    info_msg += 'is already installed, nothing to do.'
                    logger.info(info_msg % (package_name, version))
                    return True